controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -494,6 +553,15 @@ func (r *StorageClassClaimReconciler) createOrReplaceStorageClass(storageClass *
 		}
 	}
 	r.log.Info("Creating StorageClass.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
+	storageClass.OwnerReferences = []metav1.OwnerReference{
+		{
+			APIVersion:         r.storageClassClaim.APIVersion,
+			Kind:               r.storageClassClaim.Kind,
+			Name:               r.storageClassClaim.Name,
+			UID:                r.storageClassClaim.UID,
+			BlockOwnerDeletion: &_true,
+		},
+	}
```

Comment:
You cannot own a cluster-wide resource from a namespaced resource!
If you used `own` you would get an error (as it is using `SetOwnerReference` internally which preform validation before adding the ownership metadata).

Please remove this code.